Computer_Architecture/CA_3.10.3_Out-of-BoundsMemory_References_and_Buffer_Overflow.md

```
@@ -0,0 +1,108 @@
+
+C 언어는 배열 참조에 대한 어떠한 경계 체크도 하지 않는다. 지역 변수들은 경계의 관계 없이 상태 정보(레지스터 값, return 주소 등)와 함께 스택에 저장 된다. 이로 인해 배열의 요소가 배열을 벗어나는 문제가 발생하고 정보가 유출 되는 등 심각한 프로그램 에러를 야기할 수 있다.
+
+## Buffer Overflow
+
+```c
+/* 라이브러리 함수 gets()의 구현 */
+char *gets(char *s)
+{
+    int c;
+    char *dest = s;
+	while ((c = getchar()) != '\n' && c != EOF)
```

Markdownlint

Column: 1
Hard tabs

(MD010, no-hard-tabs)